README.md

```diff
@@ -186,9 +186,33 @@ export class AuthGuard extends KeycloakAuthGuard {
 
 ## HttpClient Interceptor
 
-By default all HttpClient requests will add the Authorization header in the format of: Authorization: Bearer **_TOKEN_**.
+By default, all HttpClient requests will add the Authorization header in the format of: `Authorization: Bearer **_TOKEN_**`.
 
-There is also the possibility to exclude a list of URLs that should not have the authorization header. The excluded list must be provided in the keycloak initialization. For example:
+There is also the possibility to exclude requests that should not have the authorization header. This is accomplished by implementing the `shouldAddToken` method in the keycloak initialization. For example, the configuration below will not add the token to `GET` requests that match the paths `/assets` or `/clients/public`:
+
+```ts
+await keycloak.init({
+  config: {
+    url: 'http://localhost:8080/auth',
+    realm: 'your-realm',
+    clientId: 'your-client-id',
+  },
+  shouldAddToken: (req: HttpRequest<any>): Boolean => {
```

Comment:
Since we know what should be in the request here, we should have a custom type such as:

```typescript
type ShouldAddTokenRequestPayload = {
  // ⬇ I am not sure if all these are possible, please proof and narrow the type accordingly! It would also be nice if there was a type we could use here such as KeycloakRequest["method"] to share the type definitions if something like that exists instead of using a custom union here.
  method: "get" | "post" | "put" | "delete",
  url: URL["pathname"]
}
```

Accordingly we can change the implementation to something like this:

```typescript
  // ⬇ "Boolean" is a JS constructor but "boolean" is a TS type
  shouldAddToken: (req: HttpRequest<ShouldAddTokenRequestPayload>): boolean => {
    try {
      const { method, url: path } = req;
      // ⬇ will throw if the path is not properly constructible as a valid url with the current origin and thus the `try / catch` block.
      const url = new URL(path, window.location.origin); 
      const acceptablePaths = ['/assets', '/clients/public'];
      const isGetRequest = method === "get";
      const isAcceptablePathMatch = acceptablePaths.some(pattern => pattern.test(url.pathname));
    
      return !(isGetRequest && urlMatch);
    } catch {
      return false;
    } 
  }
```

Comments in the above example should ofcourse be removed if used as they are intended for context purposes only.

WDYT?

Reply:
Sounds good to me!

Reply:
Reading through this more I'm a little confused by the intent of this. I can understand wanting the user to be able to understand what the `HttpRequest` object will have at a glance, but that type definition already exists within Angular's documentation. The generic being used here only determines the type of the request body (and some methods that relate to it), so I don't think adding a `ShouldAddTokenRequest` type would be beneficial.

As for updating the variable names, I'll adjust those. I'm not the best at choosing variable names haha.

Reply:
Considering other readme sections include the relevant imports, maybe it would make sense to include the import for `HttpRequest`?

Reply:
> Considering other readme sections include the relevant imports, maybe it would make sense to include the import for `HttpRequest`?

Yes, I agree, that would be nice.

Reply:
> Reading through this more I'm a little confused by the intent of this. I can understand wanting the user to be able to understand what the `HttpRequest` object will have at a glance, but that type definition already exists within Angular's documentation. The generic being used here only determines the type of the request body (and some methods that relate to it), so I don't think adding a `ShouldAddTokenRequest` type would be beneficial.
> 
> As for updating the variable names, I'll adjust those. I'm not the best at choosing variable names haha.

Ah, I see, if we know the shape of the body then we could create a custom type for that then but if not, `unknown` over `any` might be nice to force a type guard validation on the users part. I don't know, you can decide for that one.

The two hardest things in computer science: Cache invalidation, naming things and off by one errors 😉!

Reply:
`unknown` is a great idea, I'll update that!